src/twisted/internet/_sslverify.py

```diff
@@ -1141,9 +1177,19 @@ def clientConnectionForTLS(self, tlsProtocol):
         @return: the configured client connection.
         @rtype: L{OpenSSL.SSL.Connection}
         """
+        if self._ctx is None:
+            self._ctx = self._createContext()
+            # Note: remember client options for testing.  See
+            # twisted.internet.test.test_endpoints.tlsHostnameFromEndpoint
+            self._ctx._clientOptions = self
```

Comment:
Maybe something like this 

"remember client for testing" seemls like a task for the developers 

```suggestion
            # Note: To help with testing, we keep a reference to the TLS options.
            $ See
            # twisted.internet.test.test_endpoints.tlsHostnameFromEndpoint
            self._ctx._clientOptions = self
```

Reply:
yay, also gone